app/controllers/urls_controller.rb

```diff
@@ -0,0 +1,27 @@
+class UrlsController < ApplicationController
+
+  def index #('/')
+    @url = Url.new
+    @urls = Url.all.order(:created_at)
+  end
+
+  def create #(post '/urls')
+    @new_url = Url.new(long: params[:url][:long])
+
+      if @new_url.save
+        redirect_to '/'
+      else
+        [400, "Url not Converted"]
+        flash[:notice] = "That url was invalid"
+        redirect_to '/'
+      end
+  end
+
+  def find_long #('/:short' redirects)
```

Comment:
I would name this method `follow` or `redirect` (but `redirect` might be confusing).